src/components/Slider.tsx

```diff
@@ -0,0 +1,153 @@
+import { useRef, useState, useLayoutEffect } from "react";
+
+import arrowRight from "../assets/image-arrow-right.svg";
+import arrowLeft from "../assets/image-arrow-left.svg";
+
+export type SliderElement = {
+  title: string;
+  caption: string;
+  imageSource: string;
+};
+
+type SliderProps = {
+  value: SliderElement[];
+};
+
+export function Slider({ value }: SliderProps) {
+  const carouselRef = useRef<null | HTMLDivElement>(null);
+  const arrowRightRef = useRef<null | HTMLDivElement>(null);
+  const arrowLeftRef = useRef<null | HTMLDivElement>(null);
+
+  const [isDragStart, setIsDragStart] = useState(false);
+  const [isDragging, setIsDragging] = useState(false);
+
+  const [prevPageX, setPrevPageX] = useState(0);
+  const [prevScrollLeft, setPrevScrollLeft] = useState(0);
+  const [positionDiff, setPositionDiff] = useState(0);
+
+  const showHideIcons = () => {
+    // showing and hiding prev/next icon according to carousel scroll left value
+    if (carouselRef.current && arrowRightRef.current && arrowLeftRef.current) {
+      let scrollWidth: number =
+        carouselRef.current.scrollWidth - carouselRef.current.clientWidth;
+
+      arrowRightRef.current.style.display =
+        carouselRef.current.scrollLeft == scrollWidth ? "none" : "block";
+      arrowLeftRef.current.style.display =
+        carouselRef.current.scrollLeft == 0 ? "none" : "block";
+    }
+  };
+
+  const handleMouseDown = (
+    event: React.MouseEvent<HTMLDivElement, MouseEvent>
+  ) => {
+    setIsDragStart(true);
+    setPrevPageX(event.pageX); // horizontal coordinate of mouse click relative to the left edge of parent element
+    if (carouselRef.current) {
+      setPrevScrollLeft(carouselRef.current.scrollLeft); // current scroll starting from left of selected element
+    }
+  };
+
+  const handleMouseMove = (
+    event: React.MouseEvent<HTMLDivElement, MouseEvent>
+  ) => {
+    if (!carouselRef.current) return;
+    event.preventDefault();
+
+    // when we stop holding mouse button stop dragging
+    if (!isDragStart) {
+      return;
+    }
+
+    setIsDragging(true);
+    carouselRef.current.classList.add("dragging");
```

Comment:
Change class name conditionally in JSX based on state instead modifying DOM like that.